### ISSUE TITLE
SHELL isn't set on tty1, leading to strange GNU screen session

### ISSUE THREAD

**Issue by mika** [grml/grml#135]
SHELL isn't set on tty1, so when invoking `screen` we end up with a /bin/sh session instead of a zsh shell. Noticed with grml64-full 2018.12.

**Comment by jkirk** [comment]
FTR: Just tested with the latest daily Grml Small (`grml64-small_testing_build4033.iso`). This is still an issue.

**Comment by jkirk** [comment]
FTR: $SHELL was removed from `/etc/zshrc` in grml/grml-etc-core@a5468f8 mainly as a workaround for [Debian Bug #788819](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=788819). The bug affected `dpkg` and/or `needrestart` and was closed in the meanwhile. The workaround should not be needed anymore.

In that bug report @mika also describes an `chroot` call where the $SHELL variable is passed and causes trouble if the given shell is not installed in the chroot environment. Although I think this is a general problem, we could take a look into it.

**Comment by jkirk** [comment]
I asked myself "where or what does set the SHELL variable?" and what is it used for? This lead to login(1):

> The value for $HOME, $SHELL, $PATH, $LOGNAME, and $MAIL are set according to the appropriate fields in the password entry.

So, I'd assume the SHELL variable is set by default on all systems on tty1 (I at least can confirm that behavior by logging in on my notebook @ tty1).

As the SHELL variable is set by login(1) (and not by the shell itself - no SHELL is set our zsh configuration) I asked myself why the variable is set on all other ttys (running with and without GNU/screen) but not on tty1:

When looking into the Getty systemd units, it seems like it has something to do with the `-o` option of `agetty`:
```
grml@grml ~ % systemctl cat getty@tty1.service | grep Exec
ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear - $TERM
ExecStart=
ExecStart=-/usr/share/grml-scripts/run-welcome
grml@grml ~ % systemctl cat getty@tty2.service | grep Exec
ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear - $TERM
ExecStart=
ExecStart=-/usr/share/grml-scripts/run-screen
grml@grml ~ % systemctl cat getty@tty5.service | grep Exec
ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear - $TERM
ExecStart=
ExecStart=-/sbin/agetty -a grml %I $TERM
```

`ExecStart` in the systemd unit has a comment:

> \# The '-o' option value tells agetty to replace 'login' arguments with an
> \# option to preserve environment (-p), followed by '--' for safety, and then
> \# the entered username.

From agetty(8):
> -o, --login-options "login_options"
>               Options and arguments that  are passed to login(1). Where \u is replaced by the login name.

And back to login(1):
>        -p
>            Preserve environment.

**Comment by jkirk** [comment]
Flashback in time! We already addressed this problem quite a time ago: grml/grml#14 + grml/grml-live@48143b6 + grml/grml-live@6871972

Quoting grml/grml-live@48143b6:
> The reason for that is, that the SHELL variable is empty when called
> directly via systemd whereas SHELL=/bin/zsh when started via
> grml-runtty. I could not figure out why but a solution would be to set
> the "Environment"-option in the systemd-unit which I will propose in a
> separate commit.

TL;DR: By default login(1) is called by agetty(8) and sets the SHELL variable "according to the appropriate fields in the password entry". Our tty1 calls `run-welcome`, tty2 - tty4 call `run-screen` and tty5 - tty7 call `agetty` (with autologin `grml`). We fixed tty2 - tty4 by setting the SHELL variable via the systemd "Environment"-option (see: grml/grml-live@6871972).
So, a quick fix would be to do the same for tty1.
The correct fix would IMHO be to remove the systemd "Environment"-options and set SHELL in `run-welcome` and `run-screen` as these scripts call `/bin/zsh` in the end (and actually ignoring the password entry).

Rational:
The systemd service "Getty on %I" (`getty@.service`) calls `agetty` which "opens a tty port, prompts for a login name and invokes the /bin/login command".

Breaking down `ExecStart=-/sbin/agetty -o '-p -- \\u' --noclear - $TERM`:

getty prompts for a login name (using `echo` as login-program to demonstrate what gets passed to login(1)):
```
% agetty -o '-p -- \u' --login-program "/bin/echo" -- - 

grml64-small_testing build4033 grml pts/0

grml login: grml
-p -- grml
```

After the login name is entered, login(1) is called with the given username and prompts for the password:
```
% sudo login -p -- grml
Password:
```

SHELL variable is set:
```
root@grml ~ # echo $SHELL
/bin/zsh
root@grml ~ # echo $TERM
linux
```

Quoting relevant sections of agetty(8):
> agetty opens a tty port, prompts for a login name and invokes the /bin/login command.
> [...]
> term   The value to be used for the TERM environment variable.  This overrides whatever init(8) may have set, and is inherited by login and the shell. The default is 'vt100', or 'linux' for Linux on a virtual terminal, or 'hurd' for GNU Hurd on a virtual terminal.
> [...]
>    -o, --login-options "login_options"
>               Options and arguments that  are passed to login(1). Where \u is replaced by the login name. For example:
>    -J, --noclear
>             Do not clear the screen before prompting for the login name.  By default the screen is cleared.

**Comment by mika** [comment]
This seems to be working fine now thanks to @jkirk's changes to grml-script's run-welcome + run-screen.
Great debugging, finds and fixes - thanks! :partying_face:
